sw/device/lib/testing/test_framework/ottf.c

```diff
@@ -54,11 +54,21 @@ static void report_test_status(bool result) {
   test_status_set(result ? kTestStatusPassed : kTestStatusFailed);
 }
 
+OT_ATTR_WEAK
+bool closed_source_pre_test_hook(void)  {return true;}
+
+OT_ATTR_WEAK
+bool closed_source_post_test_hook(void) {return true;}
+
 // A wrapper function is required to enable `test_main()` and test teardown
 // logic to be invoked as a FreeRTOS task. This wrapper can be used by tests
 // that are run on bare-metal.
 static void test_wrapper(void *task_parameters) {
-  bool result = test_main();
+  // Invoke weak symbol that can be overriden in closed source code.
+  bool result = closed_source_pre_test_hook();
+  
+  // Invoke weak symbol that can be overriden in closed source code. 
+  result = result && test_main() && closed_source_post_test_hook();
```

Comment:
Nit: the following is a little more readable, while still taking advantage of C's short-circuit expression evaluation
```suggestion
  result = result && test_main();
  
  // Invoke weak symbol that can be overriden in closed source code. 
  result = result && closed_source_post_test_hook();
```